Code/Framework/AzCore/AzCore/UnitTest/TestTypes.h

```diff
@@ -44,6 +44,7 @@ namespace UnitTest
 
         void SetupAllocator(const AZ::SystemAllocator::Descriptor& allocatorDesc = {})
         {
+            AZ::Environment::Create(nullptr);
```

Comment:
I know that Esteban did a lot of work around this, so maybe get some feedback there?
@amznestebanpapp

Reply:
Environment creation/destruction should be in the ITestEnvironment
Take a look to `Code\Framework\AzCore\AzCore\UnitTest\UnitTest.h` on the `SetupEnvironment` method.

If your test doesnt have an environment setup, it should be setup in the unit test environment. I recommend that test to just use the TraceBusHook since that adds the leak detection to the test module.

Reply:
Turns out this isn't needed anymore.  I think my other change to make AP tests run as a DLL might have affected this, or someone else fixed something in dev since I made this change.  I've removed this change now.